src/main/java/org/ow2/proactive/catalog/repository/entity/KeyValueMetadataEntity.java

```diff
@@ -62,25 +62,26 @@
 @Inheritance(strategy = InheritanceType.SINGLE_TABLE)
 public class KeyValueMetadataEntity implements Serializable {
 
-    @GeneratedValue(generator = "uuid2")
-    @GenericGenerator(name = "uuid2", strategy = "uuid2")
-    @Column(name = "ID", columnDefinition = "BINARY(16)")
-    @Id
-    private UUID id;
-
     @Column(name = "KEY", nullable = false)
     protected String key;
 
     @Column(name = "VALUE", nullable = false)
     protected String value;
 
+    @SuppressWarnings("DefaultAnnotationParam")
     @Column(name = "LABEL", nullable = true)
     protected String label;
 
     @ManyToOne
     @JoinColumns({ @JoinColumn(name = "CATALOGOBJECTREVISION", referencedColumnName = "ID") })
     protected CatalogObjectRevisionEntity catalogObjectRevision;
 
+    @GeneratedValue(generator = "uuid2")
+    @GenericGenerator(name = "uuid2", strategy = "uuid2")
+    @Column(name = "ID", columnDefinition = "BINARY(16)")
+    @Id
+    private UUID id;
+
     public KeyValueMetadataEntity(String key, String value, String label) {
```

Comment:
@tobwiens This is good idea. I was already thinking about this.
If we add to @UniqueConstraint besides "CATALOGOBJECTREVISION", "KEY" 
also Label it will solve all problems.